specs/Standards/StorageManagement.md

```diff
@@ -0,0 +1,423 @@
+   total: string;
+   available: string;
+}
+
+// The below structure will be returned for the method `storage_balance_bounds`.
+// Both `min` and `max` are string representations of unsigned 128-bit integers.
+//
+// `min` is the amount of tokens required to start using this contract at all
+// (eg to register with the contract). If a new contract user attaches `min`
+// NEAR to a `storage_deposit` call, subsequent calls to `storage_balance_of`
+// for this user must show their `total` equal to `min` and `available=0` .
+//
+// A contract may implement `max` equal to `min` if it only charges for initial
+// registration, and does not adjust per-user storage over time. A contract
+// which implements `max` must refund deposits that would increase a user's
+// storage balance beyond this amount.
+type StorageBalanceBounds = {
+    min: string;
+    max: string|null;
+}
+
+/************************************/
+/* CHANGE METHODS on fungible token */
+/************************************/
+// Payable method that receives an attached deposit of Ⓝ for a given account.
+//
+// If `account_id` is omitted, the deposit MUST go toward predecessor account.
+// If provided, deposit MUST go toward this account. If invalid, contract MUST
+// panic.
+//
+// If `registration_only=true`, contract MUST refund above the minimum balance
+// if the account wasn't registered and refund full deposit if already
+// registered.
+//
+// Any attached deposit in excess of `storage_balance_bounds.max` must be
+// refunded to predecessor account.
+//
+// Returns the StorageBalance structure showing updated balances.
+function storage_deposit(
+    account_id: string|null,
+    registration_only: boolean|null
+): StorageBalance
+
+// Withdraw specified amount of available Ⓝ for predecessor account.
+//
+// This method is safe to call. It MUST NOT remove data.
+//
+// `amount` is sent as a string representing an unsigned 128-bit integer. If
+// omitted, contract MUST refund full `available` balance. If `amount` exceeds
+// predecessor account's available balance, contract MUST panic.
+//
+// If predecessor account not registered, contract MUST panic.
+//
+// MUST require exactly 1 yoctoNEAR attached balance to prevent restricted
+// function-call access-key call (UX wallet security)
+//
+// Returns the StorageBalance structure showing updated balances.
+function storage_withdraw(amount: string|null): StorageBalance
+
+// Unregisters the predecessor account and returns the storage NEAR deposit.
+//
+// If the predecessor account is not registered, the function MUST return
+// `false` without panic.
+//
+// If `force=true` the function SHOULD ignore existing account data, such as
+// non-zero balances on an FT contract (that is, it should burn such balances),
+// and close the account. Otherwise, MUST panic if caller has existing account
+// data, such as a positive registered balance (eg token holdings) or if the
+// contract doesn't support forced unregistration.
```

Comment:
Add note that if the contract implementation doesn't support forced closure of the account, the contract MAY panic. For example there are too much data to delete at once.

Reply:
Updated to

```
// If `force=true` the function SHOULD ignore existing account data, such as
// non-zero balances on an FT contract (that is, it should burn such balances),
// and close the account. Contract MAY panic if it doesn't support forced
// unregistration, or if it can't force unregister for the particular situation
// (example: too much data to delete at once).
//
// If `force=false` or `force` is omitted, the contract MUST panic if caller
// has existing account data, such as a positive registered balance (eg token
// holdings).
```